crates/oxc_linter/src/rules/react/jsx_no_target_blank.rs

```diff
@@ -0,0 +1,658 @@
+
+    #[error("eslint-plugin-react(jsx-no-target-blank): Using target=`_blank` without rel=`noreferrer` or rel=`noopener` (the former implies the latter and is preferred due to wider support) is a security risk: see https://mathiasbynens.github.io/rel-noopener/#recommendations")]
+    #[diagnostic(severity(warning), help("add rel=`noreferrer` or rel=`noopener` to the element"))]
+    NoTargetBlankWithoutNoopener(#[label] Span),
+}
+
+#[derive(Debug, Clone)]
+pub struct JsxNoTargetBlank {
+    enforce_dynamic_links: EnforceDynamicLinksEnum,
+    warn_on_spread_attributes: bool,
+    allow_referrer: bool,
+    links: bool,
+    forms: bool,
+}
+
+#[derive(Debug, Clone)]
+enum EnforceDynamicLinksEnum {
+    Always,
+    Never,
+}
+
+impl Default for JsxNoTargetBlank {
+    fn default() -> Self {
+        Self {
+            enforce_dynamic_links: EnforceDynamicLinksEnum::Always,
+            warn_on_spread_attributes: false,
+            allow_referrer: false,
+            links: true,
+            forms: false,
+        }
+    }
+}
+
+impl JsxNoTargetBlank {
+    fn diagnostic(&self, span: Span, ctx: &LintContext) {
+        if self.allow_referrer {
+            ctx.diagnostic(JsxNoTargetBlankDiagnostic::NoTargetBlankWithoutNoopener(span));
+        } else {
+            ctx.diagnostic(JsxNoTargetBlankDiagnostic::NoTargetBlankWithoutNoreferrer(span));
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule aims to prevent user generated link hrefs and form actions from creating security vulnerabilities by
+    /// requiring rel='noreferrer' for external link hrefs and form actions, and optionally any dynamically generated link
+    /// hrefs and form actions.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// When creating a JSX element that has an a tag, it is often desired to have the link open in a new tab using the
+    /// target='_blank' attribute. Using this attribute unaccompanied by rel='noreferrer', however, is a severe security
+    /// vulnerability (see noreferrer docs and noopener docs for more details) This rules requires that you accompany
+    /// target='_blank' attributes with rel='noreferrer'.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// /// correct
+    /// var Hello = <p target="_blank"></p>
+    /// var Hello = <a target="_blank" rel="noreferrer" href="https://example.com"></a>
+    /// var Hello = <a target="_blank" rel="noopener noreferrer" href="https://example.com"></a>
+    /// var Hello = <a target="_blank" href="relative/path/in/the/host"></a>
+    /// var Hello = <a target="_blank" href="/absolute/path/in/the/host"></a>
+    /// var Hello = <a></a>
+    /// /// incorrect
+    /// var Hello = <a target='_blank' href="https://example.com/"></a>
+    /// var Hello = <a target='_blank' href={dynamicLink}></a>
+    /// ```
+    JsxNoTargetBlank,
+    correctness
+);
+
+fn check_is_external_link(link: &Atom) -> bool {
+    lazy_static! {
```

Comment:
Would it be possible to replace`Regex` with something else? It may hurt performance. :(

Reply:
here doesn't seem to be a solid implementation for this at the moment, I'll make a note and add a TODO.

Reply:
What about checking whether a link starts with `"\\"` or a word?